geode-common/src/main/java/org/apache/geode/internal/inet/LocalHostUtil.java

```diff
@@ -54,70 +59,80 @@
+        try {
+          inetAddress = InetAddress.getByAddress(InetAddress.getLocalHost().getAddress());
+          if (inetAddress.isLoopbackAddress()) {
+            InetAddress ipv4Fallback = null;
+            InetAddress ipv6Fallback = null;
+            // try to find a non-loopback address
+            Set<InetAddress> myInterfaces = getMyAddresses();
+            boolean preferIPv6 = useIPv6Addresses;
+            String lhName = null;
+            for (InetAddress addr : myInterfaces) {
+              if (addr.isLoopbackAddress() || addr.isAnyLocalAddress() || lhName != null) {
+                break;
+              }
+              boolean ipv6 = addr instanceof Inet6Address;
+              boolean ipv4 = addr instanceof Inet4Address;
+              if ((preferIPv6 && ipv6) || (!preferIPv6 && ipv4)) {
+                String addrName = reverseDNS(addr);
+                if (inetAddress.isLoopbackAddress()) {
+                  inetAddress = addr;
+                  lhName = addrName;
+                } else if (addrName != null) {
+                  inetAddress = addr;
+                  lhName = addrName;
+                }
+              } else {
+                if (preferIPv6 && ipv4 && ipv4Fallback == null) {
+                  ipv4Fallback = addr;
+                } else if (!preferIPv6 && ipv6 && ipv6Fallback == null) {
+                  ipv6Fallback = addr;
+                }
+              }
             }
-          } else {
-            if (preferIPv6 && ipv4 && ipv4Fallback == null) {
-              ipv4Fallback = addr;
-            } else if (!preferIPv6 && ipv6 && ipv6Fallback == null) {
-              ipv6Fallback = addr;
+            // vanilla Ubuntu installations will have a usable IPv6 address when
+            // running as a guest OS on an IPv6-enabled machine. We also look for
+            // the alternative IPv4 configuration.
+            if (inetAddress.isLoopbackAddress()) {
+              if (ipv4Fallback != null) {
+                inetAddress = ipv4Fallback;
+                useIPv6Addresses = false;
+              } else if (ipv6Fallback != null) {
+                inetAddress = ipv6Fallback;
+                useIPv6Addresses = true;
+              }
             }
           }
+        } catch (UnknownHostException ignored) {
         }
-        // vanilla Ubuntu installations will have a usable IPv6 address when
-        // running as a guest OS on an IPv6-enabled machine. We also look for
-        // the alternative IPv4 configuration.
-        if (inetAddress.isLoopbackAddress()) {
-          if (ipv4Fallback != null) {
-            inetAddress = ipv4Fallback;
-            useIPv6Addresses = false;
-          } else if (ipv6Fallback != null) {
-            inetAddress = ipv6Fallback;
-            useIPv6Addresses = true;
-          }
-        }
-      }
-    } catch (UnknownHostException ignored) {
+        return inetAddress;
+      }, Objects::nonNull, 1, MINUTES);
```

Comment:
Why is 1 minute enough? Instead of hard coding it to 1 minute should that be the default that can be overridden by setting a system property. 
Did you intend this to be a hot retry loop? If something is trying to get DNS up it might be better not to bombard it with requests during this minute.

Reply:
you mean we should have some interval between retires?

Reply:
It certainly could. I am not to concerned about the one in thousand times this might be hit in a managed environment though. A hot loop here probably isn't the end of the world. The request will go to the local resolver first then to the DNS so its not going to be all that hot.